test/unittest/unittest_nntrainer_layers.cpp

```diff
@@ -347,6 +364,18 @@ class nntrainer_FullyConnectedLayer_TFmatch
     EXPECT_EQ(status, ML_ERROR_NONE);
   }
 
+  void loadUpdatedWeightsGradients(const char *file_uw, const char *file_g) {
+    for (int idx = 0; idx < 2; ++idx) {
+      new_w.push_back(
+        nntrainer::Tensor(layer.getParams().get()[idx].weight.getDim()));
```

Comment:
There is safer version of accessing element. `layer::paramsAt(idx)` that checks range.

Reply:
This is much better.
However, that function is protected. Made it public.

Reply:
Nice :+1: